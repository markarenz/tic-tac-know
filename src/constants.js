// STRINGS
const PLAYER = 'player';
const CPU = 'cpu';
const X = 'x';
const O = 'o';

const AI_LEVELS = {
  AI_LEVEL_RND: 'ai_level_rnd', // Always random
  AI_LEVEL_KI: 'ai_level_ki', // Killer instinct, random until a winning move is found
  AI_LEVEL_HAH: 'ai_level_hah', // Half and half, sometimes SMRT and sometimes KI
  AI_LEVEL_SMRT: 'ai_level_smrt', // Use past games to learn how to win or draw
};

const symbols = [X, O];
const sides = [PLAYER, CPU];
const coinSides = ['heads', 'tails'];
const initOpenCells = [1, 2, 3, 4, 5, 6, 7, 8, 9];
const winningPatterns = [
  [1, 2, 3], // TOP ROW
  [4, 5, 6], // MIDDLE ROW
  [7, 8, 9], // BOTTOM ROW
  [1, 4, 7], // LEFT COL
  [2, 5, 8], // MIDDLE COL
  [3, 6, 9], // RIGHT COL
  [1, 5, 9], // TL-BR DIAG
  [3, 5, 7], // TR-BL DIAG
];

const LOCAL_STORAGE_KEYS = {
  HISTORY: 'tic-tac-know-history',
  WINS_LOSSES: 'tic-tac-know-wins-losses',
};

const whoseTurnMessages = {
  player: 'Your turn',
  cpu: 'CPU Turn',
};
const gameOverMessages = {
  player: 'YOU WIN!',
  cpu: 'CPU WINS',
  draw: 'The Game is a draw.',
};

const APP_STATES = {
  GAME: 'game',
  MENU: 'menu',
};

const MENU_STATUSES = {
  PICK_SIDE: 'pick-side',
  PICK_AI: 'pick-ai',
  START: 'start',
  COMPLETE: 'complete',
};

const sideMessages = {
  x: 'You are playing as X. The CPU is playing as O.',
  o: 'You are playing as O. The CPU is playing as X.',
};
const aiLevelLabels = {
  ai_level_rnd: 'Herp Derp',
  ai_level_ki: 'Killer Instinct',
  ai_level_hah: 'Half & Half',
  ai_level_smrt: 'Mastermind',
};
const aiLevelMessages = {
  ai_level_rnd:
    'Your oppnent is, to put it bluntly, not terribly bright, making moves (and inapropriate noises) at random.',
  ai_level_ki:
    'Your opponent knows enough to be annoyingly good, but just like all your friends from college, it cannot plan ahead.',
  ai_level_hah:
    'Your opponent is playing with one chip tied behind its back, using a mix of strategies, sometimes smart and sometimes not.',
  ai_level_smrt:
    "Your opponent is smarter than you are, more disciplined and, let's face it, better looking. You don't stand a chance.",
};

//
export {
  symbols,
  sides,
  coinSides,
  PLAYER,
  CPU,
  X,
  O,
  whoseTurnMessages,
  gameOverMessages,
  sideMessages,
  aiLevelMessages,
  aiLevelLabels,
  initOpenCells,
  winningPatterns,
  AI_LEVELS,
  LOCAL_STORAGE_KEYS,
  APP_STATES,
  MENU_STATUSES,
};
