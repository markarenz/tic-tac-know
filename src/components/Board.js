import React from 'react';
import Cell from './Cell';
import { Transition } from '@tailwindui/react';
import { initOpenCells, PLAYER, CPU, whoseTurnMessages } from '../constants';
import PropTypes from 'prop-types';

const Board = ({ handlePlayerSelectCell, gameState, winsLosses }) => {
  const { whoseTurn, sideLabels, gameBoard, gameStatus, turnHistory } = gameState;
  // if (!whoseTurn) {
  //   return <div className="w-full h-screen">&nbsp;</div>;
  // }
  const result = gameStatus === 'gameOver' ? turnHistory[turnHistory.length - 1] : null;
  return (
    <div className="max-w-sm mx-auto my-4 bg-white rounded-lg p-4">
      <div className="mb-4">
        <div className="grid grid-cols-2">
          <div className="overflow-hidden">
            {whoseTurn === '' && <div> &nbsp;</div>}
            <Transition
              as="div"
              show={whoseTurn === PLAYER}
              enter="transition-all duration-500"
              enterFrom="opacity-0 -translate-x-16"
              enterTo="opacity-100 translate-x-0"
              leave="transition-all duration-500"
              leaveFrom="opacity-100 translate-x-0"
              leaveTo="opacity-0 -translate-x-16"
            >
              <div className="uppercase">
                <b>{whoseTurnMessages[PLAYER]}</b>
              </div>
            </Transition>
          </div>
          <div className="overflow-hidden">
            <Transition
              as="div"
              show={whoseTurn === CPU}
              enter="transition-all duration-500"
              enterFrom="opacity-0 translate-x-16"
              enterTo="opacity-100 translate-x-0"
              leave="transition-all duration-500"
              leaveFrom="opacity-100 translate-x-0"
              leaveTo="opacity-0 translate-x-16"
            >
              <div className="text-right uppercase">
                <b>{whoseTurnMessages[CPU]}</b>
              </div>
            </Transition>
          </div>
        </div>
      </div>

      <div className="grid grid-cols-3">
        {initOpenCells.map((i) => (
          <Cell
            key={`cell-${i}`}
            i={i}
            value={gameBoard[i]}
            whoseTurn={whoseTurn}
            gameStatus={gameStatus}
            sideLabels={sideLabels}
            handlePlayerSelectCell={handlePlayerSelectCell}
            result={result}
          />
        ))}
      </div>

      <div className="text-right mt-4">
        Wins: {winsLosses.wins}, Losses: {winsLosses.losses}, Draws: {winsLosses.draws}
      </div>
    </div>
  );
};

Board.defaultProps = {
  gameState: {
    whoseTurn: '',
  },
};
Board.propTypes = {
  gameState: PropTypes.shape({
    gameBoard: PropTypes.objectOf(PropTypes.string).isRequired,
    whoseTurn: PropTypes.string,
    sideLabels: PropTypes.objectOf(PropTypes.string).isRequired,
  }),
  handlePlayerSelectCell: PropTypes.func.isRequired,
  winsLosses: PropTypes.shape({
    wins: PropTypes.number.isRequired,
    losses: PropTypes.number.isRequired,
    draws: PropTypes.number.isRequired,
  }),
};

export default Board;
