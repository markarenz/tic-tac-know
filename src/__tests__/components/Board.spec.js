import { render, screen } from '@testing-library/react';
import Board from '../../components/Board';
import { PLAYER, CPU } from '../../constants';

const mocks = {
  handlePlayerSelectCell: jest.fn(),
  gameState: {
    whoseTurn: PLAYER,
    sideLabels: { player: 'x', cpu: 'o' },
    gameBoard: {},
    gameStatus: 'playing',
    turnHistory: [],
  },
  winsLosses: { wins: 10, losses: 5, draws: 0 },
};
describe('Board', () => {
  it('renders board in playing mode', () => {
    const container = render(
      <Board
        handlePlayerSelectCell={mocks.handlePlayerSelectCell}
        gameState={mocks.gameState}
        winsLosses={mocks.winsLosses}
      />
    );
    expect(container).toMatchSnapshot();
  });
  it('renders board in gameOver mode', () => {
    const container = render(
      <Board
        handlePlayerSelectCell={mocks.handlePlayerSelectCell}
        gameState={{
          ...mocks.gameState,
          gameStatus: 'gameOver',
        }}
        winsLosses={mocks.winsLosses}
      />
    );
    expect(container).toMatchSnapshot();
  });
  it("renders board with no one's turn", () => {
    const container = render(
      <Board
        handlePlayerSelectCell={mocks.handlePlayerSelectCell}
        gameState={{
          ...mocks.gameState,
          whoseTurn: '',
        }}
        winsLosses={mocks.winsLosses}
      />
    );
    expect(container).toMatchSnapshot();
  });
});
