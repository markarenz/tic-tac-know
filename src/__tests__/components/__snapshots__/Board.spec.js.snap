// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Board renders board in gameOver mode 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="max-w-sm mx-auto my-4 bg-white rounded-lg p-4"
      >
        <div
          class="mb-4"
        >
          <div
            class="grid grid-cols-2"
          >
            <div
              class="overflow-hidden"
            >
              <div
                class=""
                data-testid="transition-wrap"
              >
                <div
                  class="uppercase"
                >
                  <b>
                    Your turn
                  </b>
                </div>
              </div>
            </div>
            <div
              class="overflow-hidden"
            />
          </div>
        </div>
        <div
          class="grid grid-cols-3"
        >
          <div
            class="text-center border-gray-500 h-32   "
            id="cell-1"
          >
            <button
              class="w-full h-full"
              data-testid="cell-btn-1"
              disabled=""
            >
               
            </button>
          </div>
          <div
            class="text-center border-gray-500 h-32 border-l-2 border-r-2  "
            id="cell-2"
          >
            <button
              class="w-full h-full"
              data-testid="cell-btn-2"
              disabled=""
            >
               
            </button>
          </div>
          <div
            class="text-center border-gray-500 h-32   "
            id="cell-3"
          >
            <button
              class="w-full h-full"
              data-testid="cell-btn-3"
              disabled=""
            >
               
            </button>
          </div>
          <div
            class="text-center border-gray-500 h-32 border-t-2 border-b-2  "
            id="cell-4"
          >
            <button
              class="w-full h-full"
              data-testid="cell-btn-4"
              disabled=""
            >
               
            </button>
          </div>
          <div
            class="text-center border-gray-500 h-32 border-2  "
            id="cell-5"
          >
            <button
              class="w-full h-full"
              data-testid="cell-btn-5"
              disabled=""
            >
               
            </button>
          </div>
          <div
            class="text-center border-gray-500 h-32 border-t-2 border-b-2  "
            id="cell-6"
          >
            <button
              class="w-full h-full"
              data-testid="cell-btn-6"
              disabled=""
            >
               
            </button>
          </div>
          <div
            class="text-center border-gray-500 h-32   "
            id="cell-7"
          >
            <button
              class="w-full h-full"
              data-testid="cell-btn-7"
              disabled=""
            >
               
            </button>
          </div>
          <div
            class="text-center border-gray-500 h-32 border-l-2 border-r-2  "
            id="cell-8"
          >
            <button
              class="w-full h-full"
              data-testid="cell-btn-8"
              disabled=""
            >
               
            </button>
          </div>
          <div
            class="text-center border-gray-500 h-32   "
            id="cell-9"
          >
            <button
              class="w-full h-full"
              data-testid="cell-btn-9"
              disabled=""
            >
               
            </button>
          </div>
        </div>
        <div
          class="text-right mt-4"
        >
          Wins: 
          10
          , Losses: 
          5
          , Draws: 
          0
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="max-w-sm mx-auto my-4 bg-white rounded-lg p-4"
    >
      <div
        class="mb-4"
      >
        <div
          class="grid grid-cols-2"
        >
          <div
            class="overflow-hidden"
          >
            <div
              class=""
              data-testid="transition-wrap"
            >
              <div
                class="uppercase"
              >
                <b>
                  Your turn
                </b>
              </div>
            </div>
          </div>
          <div
            class="overflow-hidden"
          />
        </div>
      </div>
      <div
        class="grid grid-cols-3"
      >
        <div
          class="text-center border-gray-500 h-32   "
          id="cell-1"
        >
          <button
            class="w-full h-full"
            data-testid="cell-btn-1"
            disabled=""
          >
             
          </button>
        </div>
        <div
          class="text-center border-gray-500 h-32 border-l-2 border-r-2  "
          id="cell-2"
        >
          <button
            class="w-full h-full"
            data-testid="cell-btn-2"
            disabled=""
          >
             
          </button>
        </div>
        <div
          class="text-center border-gray-500 h-32   "
          id="cell-3"
        >
          <button
            class="w-full h-full"
            data-testid="cell-btn-3"
            disabled=""
          >
             
          </button>
        </div>
        <div
          class="text-center border-gray-500 h-32 border-t-2 border-b-2  "
          id="cell-4"
        >
          <button
            class="w-full h-full"
            data-testid="cell-btn-4"
            disabled=""
          >
             
          </button>
        </div>
        <div
          class="text-center border-gray-500 h-32 border-2  "
          id="cell-5"
        >
          <button
            class="w-full h-full"
            data-testid="cell-btn-5"
            disabled=""
          >
             
          </button>
        </div>
        <div
          class="text-center border-gray-500 h-32 border-t-2 border-b-2  "
          id="cell-6"
        >
          <button
            class="w-full h-full"
            data-testid="cell-btn-6"
            disabled=""
          >
             
          </button>
        </div>
        <div
          class="text-center border-gray-500 h-32   "
          id="cell-7"
        >
          <button
            class="w-full h-full"
            data-testid="cell-btn-7"
            disabled=""
          >
             
          </button>
        </div>
        <div
          class="text-center border-gray-500 h-32 border-l-2 border-r-2  "
          id="cell-8"
        >
          <button
            class="w-full h-full"
            data-testid="cell-btn-8"
            disabled=""
          >
             
          </button>
        </div>
        <div
          class="text-center border-gray-500 h-32   "
          id="cell-9"
        >
          <button
            class="w-full h-full"
            data-testid="cell-btn-9"
            disabled=""
          >
             
          </button>
        </div>
      </div>
      <div
        class="text-right mt-4"
      >
        Wins: 
        10
        , Losses: 
        5
        , Draws: 
        0
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Board renders board in playing mode 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="max-w-sm mx-auto my-4 bg-white rounded-lg p-4"
      >
        <div
          class="mb-4"
        >
          <div
            class="grid grid-cols-2"
          >
            <div
              class="overflow-hidden"
            >
              <div
                class=""
                data-testid="transition-wrap"
              >
                <div
                  class="uppercase"
                >
                  <b>
                    Your turn
                  </b>
                </div>
              </div>
            </div>
            <div
              class="overflow-hidden"
            />
          </div>
        </div>
        <div
          class="grid grid-cols-3"
        >
          <div
            class="text-center border-gray-500 h-32   "
            id="cell-1"
          >
            <button
              class="w-full h-full"
              data-testid="cell-btn-1"
            >
               
            </button>
          </div>
          <div
            class="text-center border-gray-500 h-32 border-l-2 border-r-2  "
            id="cell-2"
          >
            <button
              class="w-full h-full"
              data-testid="cell-btn-2"
            >
               
            </button>
          </div>
          <div
            class="text-center border-gray-500 h-32   "
            id="cell-3"
          >
            <button
              class="w-full h-full"
              data-testid="cell-btn-3"
            >
               
            </button>
          </div>
          <div
            class="text-center border-gray-500 h-32 border-t-2 border-b-2  "
            id="cell-4"
          >
            <button
              class="w-full h-full"
              data-testid="cell-btn-4"
            >
               
            </button>
          </div>
          <div
            class="text-center border-gray-500 h-32 border-2  "
            id="cell-5"
          >
            <button
              class="w-full h-full"
              data-testid="cell-btn-5"
            >
               
            </button>
          </div>
          <div
            class="text-center border-gray-500 h-32 border-t-2 border-b-2  "
            id="cell-6"
          >
            <button
              class="w-full h-full"
              data-testid="cell-btn-6"
            >
               
            </button>
          </div>
          <div
            class="text-center border-gray-500 h-32   "
            id="cell-7"
          >
            <button
              class="w-full h-full"
              data-testid="cell-btn-7"
            >
               
            </button>
          </div>
          <div
            class="text-center border-gray-500 h-32 border-l-2 border-r-2  "
            id="cell-8"
          >
            <button
              class="w-full h-full"
              data-testid="cell-btn-8"
            >
               
            </button>
          </div>
          <div
            class="text-center border-gray-500 h-32   "
            id="cell-9"
          >
            <button
              class="w-full h-full"
              data-testid="cell-btn-9"
            >
               
            </button>
          </div>
        </div>
        <div
          class="text-right mt-4"
        >
          Wins: 
          10
          , Losses: 
          5
          , Draws: 
          0
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="max-w-sm mx-auto my-4 bg-white rounded-lg p-4"
    >
      <div
        class="mb-4"
      >
        <div
          class="grid grid-cols-2"
        >
          <div
            class="overflow-hidden"
          >
            <div
              class=""
              data-testid="transition-wrap"
            >
              <div
                class="uppercase"
              >
                <b>
                  Your turn
                </b>
              </div>
            </div>
          </div>
          <div
            class="overflow-hidden"
          />
        </div>
      </div>
      <div
        class="grid grid-cols-3"
      >
        <div
          class="text-center border-gray-500 h-32   "
          id="cell-1"
        >
          <button
            class="w-full h-full"
            data-testid="cell-btn-1"
          >
             
          </button>
        </div>
        <div
          class="text-center border-gray-500 h-32 border-l-2 border-r-2  "
          id="cell-2"
        >
          <button
            class="w-full h-full"
            data-testid="cell-btn-2"
          >
             
          </button>
        </div>
        <div
          class="text-center border-gray-500 h-32   "
          id="cell-3"
        >
          <button
            class="w-full h-full"
            data-testid="cell-btn-3"
          >
             
          </button>
        </div>
        <div
          class="text-center border-gray-500 h-32 border-t-2 border-b-2  "
          id="cell-4"
        >
          <button
            class="w-full h-full"
            data-testid="cell-btn-4"
          >
             
          </button>
        </div>
        <div
          class="text-center border-gray-500 h-32 border-2  "
          id="cell-5"
        >
          <button
            class="w-full h-full"
            data-testid="cell-btn-5"
          >
             
          </button>
        </div>
        <div
          class="text-center border-gray-500 h-32 border-t-2 border-b-2  "
          id="cell-6"
        >
          <button
            class="w-full h-full"
            data-testid="cell-btn-6"
          >
             
          </button>
        </div>
        <div
          class="text-center border-gray-500 h-32   "
          id="cell-7"
        >
          <button
            class="w-full h-full"
            data-testid="cell-btn-7"
          >
             
          </button>
        </div>
        <div
          class="text-center border-gray-500 h-32 border-l-2 border-r-2  "
          id="cell-8"
        >
          <button
            class="w-full h-full"
            data-testid="cell-btn-8"
          >
             
          </button>
        </div>
        <div
          class="text-center border-gray-500 h-32   "
          id="cell-9"
        >
          <button
            class="w-full h-full"
            data-testid="cell-btn-9"
          >
             
          </button>
        </div>
      </div>
      <div
        class="text-right mt-4"
      >
        Wins: 
        10
        , Losses: 
        5
        , Draws: 
        0
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Board renders board with no one's turn 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="max-w-sm mx-auto my-4 bg-white rounded-lg p-4"
      >
        <div
          class="mb-4"
        >
          <div
            class="grid grid-cols-2"
          >
            <div
              class="overflow-hidden"
            >
              <div>
                  
              </div>
            </div>
            <div
              class="overflow-hidden"
            />
          </div>
        </div>
        <div
          class="grid grid-cols-3"
        >
          <div
            class="text-center border-gray-500 h-32   "
            id="cell-1"
          >
            <button
              class="w-full h-full"
              data-testid="cell-btn-1"
              disabled=""
            >
               
            </button>
          </div>
          <div
            class="text-center border-gray-500 h-32 border-l-2 border-r-2  "
            id="cell-2"
          >
            <button
              class="w-full h-full"
              data-testid="cell-btn-2"
              disabled=""
            >
               
            </button>
          </div>
          <div
            class="text-center border-gray-500 h-32   "
            id="cell-3"
          >
            <button
              class="w-full h-full"
              data-testid="cell-btn-3"
              disabled=""
            >
               
            </button>
          </div>
          <div
            class="text-center border-gray-500 h-32 border-t-2 border-b-2  "
            id="cell-4"
          >
            <button
              class="w-full h-full"
              data-testid="cell-btn-4"
              disabled=""
            >
               
            </button>
          </div>
          <div
            class="text-center border-gray-500 h-32 border-2  "
            id="cell-5"
          >
            <button
              class="w-full h-full"
              data-testid="cell-btn-5"
              disabled=""
            >
               
            </button>
          </div>
          <div
            class="text-center border-gray-500 h-32 border-t-2 border-b-2  "
            id="cell-6"
          >
            <button
              class="w-full h-full"
              data-testid="cell-btn-6"
              disabled=""
            >
               
            </button>
          </div>
          <div
            class="text-center border-gray-500 h-32   "
            id="cell-7"
          >
            <button
              class="w-full h-full"
              data-testid="cell-btn-7"
              disabled=""
            >
               
            </button>
          </div>
          <div
            class="text-center border-gray-500 h-32 border-l-2 border-r-2  "
            id="cell-8"
          >
            <button
              class="w-full h-full"
              data-testid="cell-btn-8"
              disabled=""
            >
               
            </button>
          </div>
          <div
            class="text-center border-gray-500 h-32   "
            id="cell-9"
          >
            <button
              class="w-full h-full"
              data-testid="cell-btn-9"
              disabled=""
            >
               
            </button>
          </div>
        </div>
        <div
          class="text-right mt-4"
        >
          Wins: 
          10
          , Losses: 
          5
          , Draws: 
          0
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="max-w-sm mx-auto my-4 bg-white rounded-lg p-4"
    >
      <div
        class="mb-4"
      >
        <div
          class="grid grid-cols-2"
        >
          <div
            class="overflow-hidden"
          >
            <div>
                
            </div>
          </div>
          <div
            class="overflow-hidden"
          />
        </div>
      </div>
      <div
        class="grid grid-cols-3"
      >
        <div
          class="text-center border-gray-500 h-32   "
          id="cell-1"
        >
          <button
            class="w-full h-full"
            data-testid="cell-btn-1"
            disabled=""
          >
             
          </button>
        </div>
        <div
          class="text-center border-gray-500 h-32 border-l-2 border-r-2  "
          id="cell-2"
        >
          <button
            class="w-full h-full"
            data-testid="cell-btn-2"
            disabled=""
          >
             
          </button>
        </div>
        <div
          class="text-center border-gray-500 h-32   "
          id="cell-3"
        >
          <button
            class="w-full h-full"
            data-testid="cell-btn-3"
            disabled=""
          >
             
          </button>
        </div>
        <div
          class="text-center border-gray-500 h-32 border-t-2 border-b-2  "
          id="cell-4"
        >
          <button
            class="w-full h-full"
            data-testid="cell-btn-4"
            disabled=""
          >
             
          </button>
        </div>
        <div
          class="text-center border-gray-500 h-32 border-2  "
          id="cell-5"
        >
          <button
            class="w-full h-full"
            data-testid="cell-btn-5"
            disabled=""
          >
             
          </button>
        </div>
        <div
          class="text-center border-gray-500 h-32 border-t-2 border-b-2  "
          id="cell-6"
        >
          <button
            class="w-full h-full"
            data-testid="cell-btn-6"
            disabled=""
          >
             
          </button>
        </div>
        <div
          class="text-center border-gray-500 h-32   "
          id="cell-7"
        >
          <button
            class="w-full h-full"
            data-testid="cell-btn-7"
            disabled=""
          >
             
          </button>
        </div>
        <div
          class="text-center border-gray-500 h-32 border-l-2 border-r-2  "
          id="cell-8"
        >
          <button
            class="w-full h-full"
            data-testid="cell-btn-8"
            disabled=""
          >
             
          </button>
        </div>
        <div
          class="text-center border-gray-500 h-32   "
          id="cell-9"
        >
          <button
            class="w-full h-full"
            data-testid="cell-btn-9"
            disabled=""
          >
             
          </button>
        </div>
      </div>
      <div
        class="text-right mt-4"
      >
        Wins: 
        10
        , Losses: 
        5
        , Draws: 
        0
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
